capstone/src/main/java/com/google/sps/servlets/ChallengeServlet.java

```diff
@@ -0,0 +1,58 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.sql.Timestamp;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.Objects.Challenge;
+
+@WebServlet("challenge")
+public class ChallengeServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Challenge").addSort("dueDate", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    // Get most recent challenge in database
+    Entity entity = results.asIterable().iterator().next();
+    Challenge challenge = Challenge.fromEntity(entity);
+    String json = new Gson().toJson(challenge);
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String challengeName = request.getParameter("name");
+    LocalDateTime dueDate = this.getDueDate(LocalDateTime.now());
+    long dueDateMillis = Timestamp.valueOf(dueDate).getTime();
+    Challenge challenge = new Challenge(challengeName, dueDateMillis, null,
+        new ArrayList<String>());
```

Comment:
https://github.com/google/google-java-format/wiki/The-Rectangle-Rule#the-rectangle-rule

```java
    Challenge challenge = new Challenge(
      challengeName,
      dueDateMillis,
      null,
      new ArrayList<String>()
    );
```

Reply:
+1